tiledb/sm/query/writers/global_order_writer.cc

```diff
@@ -1239,7 +1243,7 @@ GlobalOrderWriter::multipart_upload_state() {
     if (!state.has_value()) {
       return {Status::Ok(), {}};
     }
-    result[uri->to_string()] = std::move(*state);
+    result[uri->remove_trailing_slash().last_path_part()] = std::move(*state);
```

Comment:
Please add a comment on those 3 to explain why we are doing those string manipulations. Applies to deserialization as well. Alternatively define an inline method :)